docs/source/advanced/predicting_with_a_trained_model.md

```diff
@@ -0,0 +1,128 @@
+infer_fn = model.signatures["serving_tfrecord"]
+```
+And now to construct a dataset and get predictions on it:
+```python
+dataset = data.TFRecordDataset(glob.glob(os.path.join('/PATH/TO/DATASET', "part*")))
+total_preds =  []
+i = 0
+# A prediction loop; to predict to one batch we can simply `infer_fn(next(iter(dataset)))`
+for batch in dataset.batch(1024):
+    probs = infer_fn(protos=batch)
+    total_preds.append(probs)
+# Post processing of predictions
+```
+
+### Predicting with the default signature
+The default signature requires hitting the model with tensors. This, in turn, requires to
+do all the required preprocessing (look-ups, etc) to get these tensors.
+This is done with ml4ir. The code sceleton below describes the required steps.
+
+```python
+# Define logger
+# Define feat_config
+# Define RelevanceDataset
+# Defing RelevanceModel
+
+relevance_model.predict(relevance_dataset.test)
+```
+
+This process, while much more verbose allows to do custom pre-processing on the model
+inputs, which can be different from the preprocessing done during training.
+For images, this can be artificial blurring. For text classification, using a subset of the
+text and many others.
+
+Recall, pre-processing in ml4ir is controlled in the feature_config.yaml file.
+To do something extra during inference, we need to add it to the feature config, so that the
+pipeline is updated.
+For example, to use only the first few bytes of a text field called query that it is currently
+only preprocessed by lower-casing it, we need a function that achieves this and to pass the details
+in the config.
+So before, the features config could be:
+```bash
+    preprocessing_info:
+      - fn: preprocess_text
+        args:
+          remove_punctuation: true
+          to_lower: true
+```
+so that `preprocess_text` is the only preprocessing function. We can now do
+```bash
+    preprocessing_info:
+      - fn: preprocess_text
+        args:
+          remove_punctuation: true
+          to_lower: true
+      - fn: trim_text
+        args:
+          keep_first: 3
+```
+and define trim text in the code.
+Assuming that:
+```python
+@tf.function
+def trim_text(inp, keep_first=3):
+    """Keeps the first `keep_first` bytes of a tf.string"""
+    return tf.strings.substr(inp, 0, keep_first, unit='BYTE')
+```
+then defining the RelevanceDataset as:
+```
+relevance_dataset = RelevanceDataset(
+        data_dir="/tmp/dataset",
+        data_format=DataFormatKey.TFRECORD,
+        feature_config=feature_config,
+        tfrecord_type=TFRecordTypeKey.EXAMPLE,
+        batch_size=1024,
+        preprocessing_keys_to_fns={'trim_text': trim_text},  # IMPORTANT!
+        file_io=file_io, use_part_files=True,
+        logger=logger
+    )
+```
+will result in queries whose size is 3 bytes (as described in `trim_text`).
+
+For more information on these, [please refer to this notebook](../../../python/notebooks/predicting_with_ml4ir.ipynb).
```

Comment:
@balikasg This link will be broken. Can you use a github.com... URL perhaps? But change it to point to `master`

Reply:
yes, done now!